### PR TITLE
Update Windows feature check for environments

### LINKED ISSUE #3477: Improve Hyper-V Function Detection Based on Windows Version
### Suggested new feature or improvement

DevHome should automatically hide the Hyper-V functionality and suppress any notifications about missing Hyper-V when running on systems that do not support Hyper-V, such as Windows Home editions. This will prevent unnecessary alerts and improve the user experience.

### Scenario

DevHome should automatically detect whether Hyper-V is supported based on the Windows version. If the Windows version does not support Hyper-V (e.g., Windows Home editions), the Hyper-V functionality should be hidden.

The notification about missing or unsupported Hyper-V should be suppressed in such cases to avoid confusion and improve user experience.

### Additional details

Currently, when opening the environment page, a notification appears indicating that Hyper-V functionality is not recognized, and Hyper-V environments are not listed. This notification includes a link to enable Hyper-V functionality. Clicking the link triggers a UAC confirmation window, but it seems no action is taken, and no terminal window opens to show that the system is performing any operations. The functionality of WSL2 is unaffected, and the notification persists.

### PR DESCRIPTION
**Pull request by bbonaby**
## Summary of the pull request
This PR fixes a bug where Dev Home does not disable the Hyper-V extension when the OS SKU does not contain the Hyper-V Feature. E.g on Home sku's the querying WMI/MI for the existence of the Hyper-V feature returns the 'unknown' enum value. Before this change we only looked at if the returned enum value was 'absent'. Now we use both to check if the feature is available on the machine. For the Hyper-V extension this prevents a confusing user experience where we show the user the notification to enable Hyper-V even though it doesn't exist on their machine. 

Note: For WSL, the virtual machine platform feature is [enabled on Home SKUs](https://learn.microsoft.com/en-us/windows/wsl/faq#is-wsl-2-available-on-windows-10-home-and-windows-11-home-)

I made this generic so other places in Dev Home can use the same helper methods.

Video of what happens before this change on Home SKU (Note: the extension is enabled even though the Hyper-V feature does not exist) :

https://github.com/user-attachments/assets/2872b1fd-aa70-4dbb-a960-98c1b5e89ec4

Video of what happens after this change (Note: you can see that the extension automatically gets disabled when you attempt to query for extensions that support environments. I show the Windows feature dialog at the end to show that the Hyper-V feature does not appear in the list):

https://github.com/user-attachments/assets/db2fc291-eba9-4aa0-bd40-c44920eff93d

## References and relevant issues
Note: #3381 , in the future I have a plan to move this helper class into a service that can be used by Dev Home and the extensions as WMI/MI is being used more and more throughout the app.

When fixing this I also noticed that the provider headers appear even when there are no environments for these providers. This is a separate issue and I've created this GitHub issue: #3490 
## PR checklist
- [x] Closes #3477
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

File: common/Helpers/ManagementInfrastructureHelper.cs [hunk 1]
```diff
@@ -80,5 +87,48 @@ private static FeatureAvailabilityKind GetAvailabilityKindFromState(uint state)
...
+    /// 2. The classId is an internal Dev Home extension and the feature is either enabled or disabled.
+    /// </returns>
+    public static bool IsWindowsOptionalFeatureAvailableForExtension(string extensionClassId)
+    {
+        if (ExtensionToFeatureNameMap.TryGetValue(extensionClassId, out var featureName))
+        {
+            return IsWindowsOptionalFeatureAvailable(featureName);
+        }
+
+        // This isn't an internal Dev Home extension that we know about, so don't try to disable it.
```
Comment (krschau) [inline common/Helpers/ManagementInfrastructureHelper.cs:131]: I'm confused by this comment -- disabling it?

File: common/Services/IExtensionService.cs [hunk 2]
```diff
@@ -27,4 +27,12 @@ public interface IExtensionService
     public void EnableExtension(string extensionUniqueId);
 
     public void DisableExtension(string extensionUniqueId);
+
+    /// <summary>
+    /// Gets a boolean indicating whether the extension was disabled due to its corresponding the Windows optional feature
```
Comment (krschau) [inline common/Services/IExtensionService.cs:32]:
```suggestion
    /// Gets a boolean indicating whether the extension was disabled due to the corresponding Windows optional feature
```

File: src/Services/ExtensionService.cs [hunk 3]
```diff
@@ -380,4 +390,25 @@ public void DisableExtension(string extensionUniqueId)
...
+    /// <inheritdoc cref="IExtensionService.DisableExtensionIfWindowsFeatureNotAvailable(IExtensionWrapper)"/>
+    public async Task<bool> DisableExtensionIfWindowsFeatureNotAvailable(IExtensionWrapper extension)
+    {
+        // Only attempt to disable feature if its available.
+        if (IsWindowsOptionalFeatureAvailableForExtension(extension.ExtensionClassId))
+        {
+            return false;
+        }
+
+        _log.Information($"Disabling extension: '{extension.ExtensionDisplayName}' because its feature is absent or unknown");
```
[the remark this answers is not in the document]
Comment (bbonaby) [reply src/Services/ExtensionService.cs]: that's fair, I updated this to use the `Warning` instead

File: common/Helpers/ManagementInfrastructureHelper.cs [hunk 4]
```diff
@@ -22,7 +24,12 @@ public static class ManagementInfrastructureHelper
 {
     private static readonly ILogger _log = Log.ForContext("SourceContext", nameof(ManagementInfrastructureHelper));
 
-    public static FeatureAvailabilityKind IsWindowsFeatureAvailable(string featureName)
+    public static readonly Dictionary<string, string> ExtensionToFeatureNameMap = new()
+    {
+        { HyperVExtensionClassId, HyperVWindowsOptionalFeatureName },
```
[the remark this answers is not in the document]
Comment (bbonaby) [reply common/Helpers/ManagementInfrastructureHelper.cs]: updated this and used a list instead. Thanks

File: common/Services/ComputeSystemService.cs [hunk 5]
```diff
@@ -61,13 +61,11 @@ public async Task<List<ComputeSystemProviderDetails>> GetComputeSystemProvidersA
                     continue;
                 }
 
-                // If we're looking at the Hyper-V extension and the feature isn't present on the users machine, disable the extension.
-                // This can happen if the user is not on a SKU that supports Hyper-V.
-                if (extension.ExtensionClassId.Equals(CommonConstants.HyperVExtensionClassId, StringComparison.OrdinalIgnoreCase) &&
-                    ManagementInfrastructureHelper.IsWindowsFeatureAvailable(CommonConstants.HyperVWindowsOptionalFeatureName) == FeatureAvailabilityKind.Absent)
+                // If we're looking at an internal extension that relies on a Windows optional feature being present on the machine. We need to
+                // check if its on the machine first before allowing it to be added to the list of extensions that are retrieved. If the feature is
+                // absent or unknown we disable the extension.
```
Comment (huzaifa-d) [inline common/Services/ComputeSystemService.cs:66]:
```suggestion
                // absent or unknown, we disable the extension.

```

File: common/Services/ComputeSystemService.cs [hunk 6]
```diff
@@ -61,13 +61,11 @@ public async Task<List<ComputeSystemProviderDetails>> GetComputeSystemProvidersA
                     continue;
                 }
 
-                // If we're looking at the Hyper-V extension and the feature isn't present on the users machine, disable the extension.
-                // This can happen if the user is not on a SKU that supports Hyper-V.
-                if (extension.ExtensionClassId.Equals(CommonConstants.HyperVExtensionClassId, StringComparison.OrdinalIgnoreCase) &&
-                    ManagementInfrastructureHelper.IsWindowsFeatureAvailable(CommonConstants.HyperVWindowsOptionalFeatureName) == FeatureAvailabilityKind.Absent)
+                // If we're looking at an internal extension that relies on a Windows optional feature being present on the machine. We need to
+                // check if its on the machine first before allowing it to be added to the list of extensions that are retrieved. If the feature is
```
Comment (huzaifa-d) [inline common/Services/ComputeSystemService.cs:65]:
```suggestion
                // check if it's on the machine first before allowing it to be added to the list of extensions that are retrieved. If the feature is

```

**Review by huzaifa-d — APPROVED** [review]

**Review by sshilov7 — APPROVED** [review]

**Pull request merged**